blob/blob.go

```diff
@@ -1073,3 +1093,9 @@ func wrapError(b driver.Bucket, err error) error {
 }
 
 var errClosed = gcerr.Newf(gcerr.FailedPrecondition, nil, "blob: Bucket has been closed")
+
+// PrefixedBucket returns a *Bucket based on b with all keys modified to have
+// prefix.
+func PrefixedBucket(b *Bucket, prefix string) *Bucket {
+	return NewBucket(driver.NewPrefixedBucket(b.b, prefix))
```

Comment:
Is it necessary to implement this in terms of a `driver.Bucket`? I am concerned that this implies that our portable type is not good at composing, since folks may want to create similar layers and wouldn't have access to the driver of an arbitrary portable type instance.

Reply:
Let's chat in person.

-- No, it is not necessary, but I think you necessarily end up with a different type. I.e., `type PrefixedBucket` for the wrapped type, not `blob.Bucket`.
-- That makes it hard to use a possibly-prefixed bucket.
-- In particular, I'm not sure if it composes well with an `URLOpener` that returnes a prefixed bucket.